### DISCUSSION TITLE
Allow customization of foreign key type

### DISCUSSION THREAD

**Discussion by mortenscheel** [cjmellor/level-up#23] [Ideas]
I've tried the package in two different projects, and the migrations failed for different reasons.

In version 5.8, Laravel changed the default primary keys from `INT USIGNED` to `BIGINT UNSIGNED` So in projects that started before 5.8, `users.id` is most likely not a bigint.
```php
// This fails if users.id is not an unsigned bigint
$table->foreignId(config('level-up.user.foreign_key'))->constrained(config('level-up.user.users_table'));

// This would work
$table->unsignedInteger(config('level-up.user.foreign_key'));
$table->foreign(config('level-up.user.foreign_key'))->references('id')->on(config('level-up.user.users_table'));
```

And in a newer project I'm using UUID as primary key on users, in which case you'd have to use this in the migrations:
```php
$table->foreignUuid(config('level-up.user.foreign_key'))->constrained(config('level-up.user.users_table'));
```

It's not a huge hassle to edit the migrations manually, but since you already allow some degree of customization of the foreign key, it would be cool if the data type could be customized in config as well.

**Comment by cjmellor** [comment]
Hi,

I'm open to the idea. Is it something you could PR? Not sure when I'd be able to look into it myself. 

Thanks

**Reply by mortenscheel** [reply]
Sure. I think I'll have time next week.

**Comment by cjmellor** [comment]
One thing I forgot to mention was; if you publish the migrations you can just make these changes yourself to suit your needs -- if that's easier.

**Reply by mortenscheel** [reply]
Yeah I know. It's easy enough if you understand what the problem is and how to solve it.
